web/v1/snowflake/sql-runner/sql/standard/04-users/01-main/02-users-sessions-this-run.sql

```diff
@@ -0,0 +1,31 @@
+/*
+   Copyright 2021 Snowplow Analytics Ltd. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+
+CREATE OR REPLACE TABLE {{.scratch_schema}}.users_sessions_this_run{{.entropy}}
+AS (
+    SELECT
+        a.*
+
+    FROM
+        {{.output_schema}}.sessions{{.entropy}} AS a
+
+    INNER JOIN {{.scratch_schema}}.users_userids_this_run{{.entropy}} AS b
+        ON a.domain_userid = b.domain_userid
+
+    WHERE a.start_tstamp >= (SELECT MIN(start_tstamp) FROM {{.scratch_schema}}.users_userids_this_run{{.entropy}})
```

Comment:
Why use nested query in this step instead of limits table?

Reply:
From my tests, this was slightly faster. No other reason.

Reply:
Ah this makes sense. So one thing that I wonder is whether we should still have the limits tables, to accommodate custom steps.

The thing about the users step is that we _don't_ stage data for this level of aggregation ahead of time, because it would be too much data. So I'm not sure on what we should do here - if people are likely to want to query the sessions table in order to aggregate to the users level, we should provide them with limits to use. 

Does this make sense?

Reply:
Yes, it certainly makes sense. Maybe we could also then add that table as a temporary output of the users module in the readme's too, so that is clear that it should exist and it can potentially be used?

Reply:
Sorry missed your last comment. Yes I think that's a good idea!